cloudomate/util/captchasolver.py

```diff
@@ -0,0 +1,153 @@
+import json, requests
+from time import sleep
+import os
+import base64
+
+"""
+use this already generated API key for testing: fd58e13e22604e820052b44611d61d6c
+
+Usage: 
+
+Feed Anti Captcha account API key to solver
+c_solver = captchaSolver("fd58e13e22604e820052b44611d61d6c")
+
+Find out how much money us left
+temp = c_solver.getBalance()
+
+Provide captcha image full path to solver method to get solution
+solution = c_solver.solveCaptchaTextCaseSensitive("/home/testm2/Desktop/testcscr/php/pyhton/cap2.jpg")
+print(str(solution))
+"""
+
+class captchaSolver():
+
+    __clientKey = "not set"
+         
+    def __init__(self, cKey):
+        self.__clientKey = cKey
+
+    def getBalance(self):
+        # Query API for account balance
+        r = requests.post('https://api.anti-captcha.com/getBalance',
+                          json={"clientKey": self.__clientKey})
+
+        # Check response of HTTP request
+        if(r.status_code == requests.codes.ok):
+            j = json.loads(r.text)
+            if(j["errorId"] == 0):
+                print("Successful, account balance returned")
+                return j["balance"]
+            else:
+                # Handle API error
+                print(r.text)
+        else:
+            # Handle request error
+            print(r.status_code)
+
+    def solveCaptchaTextCaseSensitive(self, fullImageFilePath):
+        # Encode captcha image before sending it
+        encoded_image_string = ""
+        if(os.path.isfile(fullImageFilePath) == True):
+            with open(fullImageFilePath, "rb") as image_file:
+                encoded_image_string = base64.b64encode(image_file.read())
+                print("Captcha image sucessfully encoded")
+        else:
+            print("Error: file path not found")
+            return
+
+        # Create new Captcha solving task using the API
+        taskId = self.__createTaskCaptchaTextCaseSensitive(encoded_image_string)
+        
+        # Query API until task is finished
+        currentStatus = self.__getTaskStatus(taskId)
+        while currentStatus == "processing":
+            print("Sleeping 5 sec")
+            sleep(5)
+            currentStatus = self.__getTaskStatus(taskId)
+            print("Current status: " + str(currentStatus))
+
+        # Get solution
+        solution = self.__getTaskResult(taskId)
+        return solution["text"]
+
```

Comment:
Please remove a few newlines here (the default is to have one newline between methods).